aten/src/ATen/NumericUtils.h

```diff
@@ -51,5 +51,56 @@ inline C10_HOST_DEVICE bool _isnan(at::BFloat16 val) {
   return at::_isnan(float(val));
 }
 
+template <typename T,
+          typename std::enable_if<std::is_same<T, double>::value, int>::type = 0>
+C10_HOST_DEVICE inline T exp(T x) {
+  return ::exp(x);
```

Comment:
A little surprised you didn't need to condition on nvcc to swap between `std::exp` and `::exp`